src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -129,6 +132,29 @@ private HydrationTransformation getTransformationForField(List<HydrationTransfor
         return transformations.stream().filter(transformation -> transformation.getNewField() == field).findFirst().get();
     }
 
+    private CompletableFuture<RootExecutionResultNode> callService(ExecutionContext executionContext,
+                                                                   QueryTransformationResult queryTransformerResult,
+                                                                   Service service,
+                                                                   Operation operation) {
+
+        Map<Field, FieldTransformation> transformationByResultField = queryTransformerResult.getTransformationByResultField();
+        List<MergedField> transformedMergedFields = queryTransformerResult.getTransformedMergedFields();
+
+        ServiceExecution serviceExecution = service.getServiceExecution();
+        GraphQLSchema underlyingSchema = service.getUnderlyingSchema();
+
+        ServiceExecutionParameters serviceExecutionParameters = buildServiceExecutionParameters(executionContext, queryTransformerResult);
+        ExecutionContext executionContextForService = buildServiceExecutionContext(executionContext, underlyingSchema, serviceExecutionParameters);
+
+        ExecutionStepInfo rootExecutionStepInfo = createRootExecutionStepInfo(service.getUnderlyingSchema(), operation);
+
+        CompletableFuture<ServiceExecutionResult> result = serviceExecution.execute(serviceExecutionParameters);
+        assertNotNull(result, "A service execution MUST provide a non null CompletableFuture<ServiceExecutionResult> ");
```

Comment:
what would happen with execution? I mean where this exception if it happens will bubble up? should you returned CompletableFuture with error here instead of throwing exception (by asserting not null) directly?